lib/screens/lock/lock_screen.dart

```diff
@@ -0,0 +1,302 @@
+import 'package:deus/core/widgets/text_field_with_max.dart';
+import 'package:deus/statics/my_colors.dart';
+import 'package:deus/statics/styles.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_countdown_timer/flutter_countdown_timer.dart';
+
+enum LockStates {
+  hasToApprove,
+  pendingApprove,
+  canLock,
+  lockPendingApprove,
+  lockApproved
+}
+
+class LockScreen extends StatefulWidget {
+  static const url = '/Lock';
+
+  @override
+  _LockScreenState createState() => _LockScreenState();
+}
+
+class _LockScreenState extends State<LockScreen> {
+  final String address = '0x3a6dabD6B5C75291a3258C29B418f5805792a87e';
+
+
+  final balance = 13.234;
+
+  final gradient = MyColors.blueToGreenGradient;
+
+  LockStates _lockState = LockStates.hasToApprove;
+  bool _showToast = false;
+
+  final _textController = TextEditingController();
+
+  int endTime = DateTime.now().millisecondsSinceEpoch + 1000 * 30; // current Time in the locking period
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      backgroundColor: Color(MyColors.Background),
+      body: SafeArea(
+        child: SingleChildScrollView(
+          padding: const EdgeInsets.symmetric(vertical: 20, horizontal: 15),
+          child: SizedBox(
+            height: MediaQuery.of(context).size.height + 50,
+            child: Column(
+              mainAxisSize: MainAxisSize.min,
+              crossAxisAlignment: CrossAxisAlignment.start,
+              children: [
+                HeaderWithAddress(
+                  walletAddress: address,
+                ),
+                SizedBox(
+                  height: 23,
+                ),
+                BackButtonWithText(),
+                SizedBox(
+                  height: 20,
+                ),
+                Text(
+                  'Lock your DEA',
+                  style: TextStyle(fontSize: 25),
+                ),
+                SizedBox(
+                  height: 20,
+                ),
+                Text(
+                  'If you lock your DEA you will mint sDEA which can be used for single asset staking or for liquidity pools.',
+                  style: MyStyles.whiteMediumTextStyle,
+                ),
+                CountdownTimer(
+                  endTime: endTime,
+                  widgetBuilder: (_, time) {
+                    if (time == null) {
+                      return Text('Current locking period still lasts 0 Days, 0 Hours, 0 Minutes and 0 Seconds.', style: MyStyles.gradientMediumTextStyle);
+                    } else {
+                      return Text(
+                        'Current locking period still lasts ${time.days != null ? time.days : 0} Days, ${time.hours != null ? time.hours : 0} Hours, ${time.min != null ? time.min : 0} Minutes and ${time.sec != null ? time.sec : 0} Seconds.',
+                        style: MyStyles.gradientMediumTextStyle,
+                      );
+                    }
```

Comment:
Bitte überprüf einmal, was passiert, wenn die endTime in der Vergangeheit liegt. Steht dann eine negative Zahl dran, also z.B. lasts -2 days oder steht dann 0 Tage dran? Kann mir irgendwie nicht vorstellen, dass time = null ist, wenn er abgelaufen ist. Oder ist das so?

- [ ] Bitte kurz kommentieren, ob es auch 0 Tage anzeigt, wenn die endTime in der Vergangenheit ist.

Reply:
Also das kommentieren bezieht sich hier auf dieses Github Pullrequest, also schreib mir kurz, ob es klappt.

- [ ] Aber im Code dokumentieren macht auch Sinn, also schreib nen kurzen Codekommentar, was es bedeutet, wenn time = null ist. Tritt das nur ein, wenn die endTime null ist oder ist es nur null, falls die Zeit abgelaufen ist?

Reply:
wenn die DateTime in der vergangenheit liegt ist der Timer einfach bei allen bei 0